doc/specs/Proto extensions-spec.md

```diff
@@ -38,66 +38,178 @@ in between 2 and 3:
+
+```js
+{
+    "guid": "{b453ae62-4e3d-5e58-b989-0a998ec441b8}",
+    "fontSize": 16,
+    "fontWeight": "thin"
+}
+```
+
+**NOTE**: This will *not* change the way the profile looks in the user's settings file. The Azure cloud shell profile
+in their settings file (assuming they have made no changes) will still be as below. 
+
+```js
+{
+    "guid": "{b453ae62-4e3d-5e58-b989-0a998ec441b8}",
+    "hidden": false,
+    "name": "Azure Cloud Shell",
+    "source": "Windows.Terminal.Azure"
+}
+```
+However, the user is free to make changes to it as usual and those changes will have the 'final say'. 
 
 #### Full profile stubs
 
 Technically, full profile stubs do not need to contain anything (they could just be '\{\}'). However we should
 have some qualifying minimum criteria before we accept a stub as a full profile. I suggest that we only create
 new profile objects from stubs that contain at least the following
 
-* A name (and this name _should not_ match the names of existing profiles as described above, because that would cause us to interpret this profile as modifying an existing one)
+* A name
 * A commandline argument
 * A unique GUID - this is so we avoid conflicts if several different creators name their profile the same thing
 
 As in the case of the dynamic profile generator, if we create a profile that did not exist before (i.e. does not
-exist in the user settings), we need to add the profile to the user settings file and re-save that file. 
+exist in the user settings), we need to add the profile to the user settings file and re-save that file.
+
+Furthermore, we will add a source field to profiles created this way (again, similar to what we do for dynamic profiles).
+The source field value is dependent on how we obtained this json file, and will be touched on in more detail later in the
+spec (see "Creation and location(s) of the json files" - "The source field"). 
+
+Here is an example of a json stub that contains a full profile:
+
+```js
+{
+    "guid": "{a821ae62-9d4a-3e34-b989-0a998ec283e6}",
+    "name": "Cool Profile",
+    "commandline": "powershell.exe",
+    "antialiasingMode": "aliased",
+    "fontWeight": "bold",
+    "scrollbarState": "hidden"
+}
+```
+
+When this profile gets added back to the user's settings file, it will look similar to the way we currently add
+new dynamic profiles to the user's settings file. Going along with the example above, the corresponding
+json stub in the user's settings file will be:
+
+```js
+{
+    "guid": "{a821ae62-9d4a-3e34-b989-0a998ec283e6}",
+    "name": "Cool Profile",
+    "hidden": "false",
+    "source": "local"
+}
+```
+Again, the user will be able to make changes to it as they do for all other profiles. 
+
+#### Creation of colour schemes
+
+As with full profiles, we will have some qualifying criteria for what we accept as full colour schemes: color schemes
+must have
+* A name
+* A background
+* A foreground

```

Comment:
Personally, don't like the idea of partially defined color schemes. I'd say every field is required (except selectionBackground?).

Especially because foreground/background can just be defined on a profile (stub).

Reply:
I'm not sure about needing to specify _every_ field though - if someone wants to create a fairly simple colour scheme I don't think they should need to specify BrightCyan, BrightGreen etc. 

How about, at the minimum, they should specify: foreground, background, cursor colour and selection background?